utils/filter.c

```diff
@@ -836,6 +838,16 @@ static void setup_trigger(char *filter_str, struct symtabs *symtabs,
 
 		/* use demangled name for triggers (some auto-args need it) */
 		name = demangle(name);
+
+		if (flags & TRIGGER_FL_LOC) {
+			if (!strpbrk(name, REGEX_CHARS) && is_directory(name)) {
+				re_suffix = name[strlen(name) - 1] == '/' ? ".*" : "/.*";
+				xasprintf(&tmp, "%s%s", name, re_suffix);
```

Comment:
Actually you can just add a slash since kernel can handle double slashes.  Also you need to check pattern type as well - it'd be '*' for glob patterns.

Reply:
Hmm.. we cannot simply check if it's a directory since the directory can be in a different location or does not exist at all.  Note that this is a name of directory at the compile time and we are checking it at runtime.